### PR TITLE
Ddce 872:Not Eligible 1 Page

### PR DESCRIPTION
**Pull request by prameya-phatak**
# DDCE-872

##New feature 

Not Eligible 1 Page

## PR Suggestions
- Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?
- Have you done a visual check of the changes?
- Is there any still commented or unused code? Lingering printlns?
- Have things been implemented in a complicated way?
- Are the test still over the coverage threshold?
- Does the compiler throw a lot of warnings? 
- Have methods and tests been named clearly?
- Were there any changes in the config? Do changes need to be made in app-config-???

## Checklist PR Raiser
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

## Checklist PR Reviewer
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [ ]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: app/controllers/checkEligibility/InEligibleCharitablePurposesController.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    val sessionRepository: SessionRepository,
+    val navigator: EligibilityNavigator,
+    identify: SessionIdentifierAction,
+    getData: DataRetrievalAction,
+    requireData: DataRequiredAction,
+    val controllerComponents: MessagesControllerComponents,
+    view: InEligibleCharitablePurposesView
+  )(implicit appConfig: FrontendAppConfig) extends LocalBaseController {
+
+  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) { implicit request =>
```
Comment (anburoach) [inline app/controllers/checkEligibility/InEligibleCharitablePurposesController.scala:39]: mode is not using anywhere here... so plz remove it

File: app/views/checkEligibility/InEligibleCharitablePurposesView.scala.html [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import controllers.checkEligibility.routes._
+@import models.Mode
```
Comment (anburoach) [inline app/views/checkEligibility/InEligibleCharitablePurposesView.scala.html:18]: Remove the Mode

File: app/views/checkEligibility/InEligibleCharitablePurposesView.scala.html [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+@import views.ViewUtils._
+@import config.FrontendAppConfig
+
+@this(
+    govukLayout: templates.GovukLayoutWrapper,
+    h1: components.h1,
+    button: components.button,
```
Comment (anburoach) [inline app/views/checkEligibility/InEligibleCharitablePurposesView.scala.html:24]: button is nowhere used in this file so plz remove

File: app/views/checkEligibility/InEligibleCharitablePurposesView.scala.html (hunk 2 above)
Comment (prameya-phatak) [reply app/views/checkEligibility/InEligibleCharitablePurposesView.scala.html]: removed.

File: app/views/checkEligibility/InEligibleCharitablePurposesView.scala.html [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+@()(implicit request: Request[_], messages: Messages, appConfig: FrontendAppConfig)
+
+@govukLayout(pageTitle = Some(titleNoForm("inEligibleCharitablePurposes.title", None)), signOut = false) {
+
+    @h1(messages("inEligibleCharitablePurposes.heading"))
+
+    @p(Html(messages("inEligibleCharitablePurposes.p1")))
+    @p(Html(messages("inEligibleCharitablePurposes.p2")))
+    <p class="govuk-body">@messages("inEligibleCharitablePurposes.p3") <a href="https://www.gov.uk/charities-and-tax/get-recognition" target="_blank">@messages("inEligibleCharitablePurposes.p3.link")</a>
```
Comment (anburoach) [inline app/views/checkEligibility/InEligibleCharitablePurposesView.scala.html:37]: use link component for 'a' tag

Comment (prameya-phatak) [reply app/views/checkEligibility/InEligibleCharitablePurposesView.scala.html]: modified

File: app/views/checkEligibility/InEligibleCharitablePurposesView.scala.html (hunk 3 above)
Comment (prameya-phatak) [reply app/views/checkEligibility/InEligibleCharitablePurposesView.scala.html]: removed

File: app/controllers/checkEligibility/InEligibleCharitablePurposesController.scala (hunk 1 above)
Comment (prameya-phatak) [reply app/controllers/checkEligibility/InEligibleCharitablePurposesController.scala]: removed

**Pull request merged**